packages/core/src/index.ts

```diff
@@ -270,50 +271,51 @@ export async function processManifest(url: string, scope: string, module: string
       method: 'GET',
       headers,
     });
-    const response = await manifestPromise;
-
-    // handle network errors
-    if (!response.ok) {
-      let error = 'Unable to process manifest';
-      const resClone = response.clone();
-      try {
-        error = await resClone.json();
-      } catch {
-        // try text error if json fails
-        error = await resClone.text();
+    return manifestPromise.then((response) => {
```

Comment:
Wouldn't it better to wrap this within `try`/`catch` clause to propagate these errors correctly?

Reply:
The issue is the parent `new Promise`. That can't have an inner async function. Errors will be lost.

And yeah it also includes any inner async/await calls. So if we want to be 100% sure our inner errors are not lost, we have to do it this way.

Before, if manifest fetch resulted in 404, you would get an infinite loading state in scalprum component as the promise will not get properly resolved. Even though we were doing try/catch in the correct places.

I don't like this either but I don't think we have any other option.